tests/e2e/test_graphql_client.py

```diff
@@ -0,0 +1,52 @@
+"""
+Test module for the GraphQL client
+"""
+from unittest import mock
+
+import pytest
+
+from kili.client import Kili
+from kili.exceptions import GraphQLError, TransportQueryError
+
+
+@pytest.mark.parametrize(
+    "query",
+    [
+        "query MyQuery { my_query_is_clearly_not_valid_according_to_the_kili_schema { id } }",
+        "query { projects { this_field_does_not_exist } }",
+    ],
+)
+def test_gql_bad_query_local_validation(query):
+    """test gql validation against local schema"""
+    kili = Kili()  # fetch schema from kili server
+
+    mocked_transport_execute = mock.MagicMock()
+    kili.auth.client._gql_client.transport.execute = mocked_transport_execute  # type: ignore
+
+    with pytest.raises(GraphQLError):
+        kili.auth.client.execute(query)
+
+    mocked_transport_execute.assert_not_called()
+
+
+def test_gql_bad_query_remote_validation():
+    """test validation by the server no local schema"""
+    kili = Kili()
+
+    query = """
+    query MyQuery {
+        my_query_is_clearly_not_valid_according_to_the_kili_schema {
+            id
+        }
+    }
+    """
+
+    kili.auth.client._gql_client.schema = None
+    kili.auth.client._gql_client.fetch_schema_from_transport = False
+    mocked_validate = mock.MagicMock()
+    kili.auth.client._gql_client.validate = mocked_validate
+
+    with pytest.raises(TransportQueryError, match="Cannot query field"):
+        kili.auth.client.execute(query)
+
+    mocked_validate.assert_not_called()
```

Comment:
What is the difference between the 2 cases ?

Reply:
one is local query validation, one is remote query validation by the backend

Reply:
I don't get the difference between the two. When do we do a local validation or backend validation?
Can we choose when to validate locally or in the backend? And are there reasons to rather do it locally or in the backend?

Reply:
local validation checks for correct query format

backend validation checks for query format and other rights (cannot query assets if no project id, etc.)

Reply:
we can choose not to do it locally yes

we can do it locally since it doesn't cost anything more to do it locally

can be used for testing purposes

Reply:
Thanks!

Reply:
+ the response time should be faster if it is first done locally